core/src/test/scala/org/apache/spark/sql/delta/MergeIntoSQLSuite.scala

```diff
@@ -275,4 +275,21 @@ class MergeIntoSQLSuite extends MergeIntoSuiteBase  with DeltaSQLCommandTest {
       }
     }
   }
+
+  test("Negative case - non-delta target") {
+    withTable("source", "target") {
+      Seq((1, 1), (0, 3), (1, 5)).toDF("key1", "value").createOrReplaceTempView("source")
+      Seq((1, 1), (0, 3), (1, 5)).toDF("key2", "value").write.saveAsTable("target")
+
+      val e = intercept[UnsupportedOperationException] {
+        executeMerge(
+          target = "target",
+          source = "source src",
+          condition = "src.key1 = target.key2",
+          update = "key2 = 20 + key1, value = 20 + src.value",
+          insert = "(key2, value) VALUES (key1 - 10, src.value + 10)")
+      }.getMessage
+      errorContains(e, "MERGE INTO TABLE is not supported temporarily")
```

Comment:
what is the error message otherwise? just curious how do things fail now without this change?

Reply:
Without the change the message it's an MERGE destination only supports Delta sources. This message it's comming from spark https://github.com/apache/spark/blob/dc7b5a99f07e172de2c7530e6a7c14dcf6e4f217/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala#L727
As Spark strategies it's the last one to be executed by spark, be removing this. If you add for exemple Iceberg Extensions, if it's an iceberg table, it extensions will take over